docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,298 @@
+# V2 Migration Guide
+
+!!! edgey "EdgeX 2.0"
+    For the EdgeX 2.0 (Ireland) release there are many backward breaking changes. These changes require custom Application Services and custom profiles (app-service-configurable) to be migrated. This section outlines the necessary steps for this migration.
+
+## Custom Application Services
+
+### Configuration
+
+The migration of any Application Service's configuration starts with migrating configuration common to all EdgeX services. See the [V2 Migration of Common Configuration](../../configuration/V2MigrationCommonConfig) section for details. The remainder of this section focuses on configuration specific to Application Services.
+
+#### SecretStoreExclusive
+
+The `SecretStoreExclusive` section has been removed in EdgeX 2.0. With Edgex 2.0 all SecretStores are exclusive, so the existing `SecretStore` section is all that is required. Services requiring `known secrets` such as `redisdb` must inform the `Security SecretStore Setup` service (via environment variables) that the application service requires the secret added to its SecretStore. See the [Configuring Add-on Service](../../../security/Ch-Configuring-Add-On-Services) section for more details.
+
+#### Clients
+
+The client used for the version validation check has changed to being from Core Metadata, rather than Core Data. This is because Core Data is now optional when persistence isn't required since all Device Services are publishing directly to the EdgeX MessageBus. The configuration for Core Metadata is the only `Clients` entry required, all other (see below) are optional based on use case needs. 
```

Comment:
```suggestion
The client used for the version validation check has changed to being from Core Metadata, rather than Core Data. This is because Core Data is now optional when persistence isn't required since all Device Services publish directly to the EdgeX MessageBus. The configuration for Core Metadata is the only `Clients` entry required, all other (see below) are optional based on use case needs. 
```